src/sentry/utils/redis.py

```diff
@@ -198,22 +199,26 @@ def cluster_factory() -> (
         # losing some type safety: SimpleLazyObject acts like the underlying type
         return SimpleLazyObject(cluster_factory)  # type: ignore[return-value]
 
-    def get(self, key: str) -> RedisCluster[str] | StrictRedis[str]:
-        try:
-            return self._clusters_str[key]
-        except KeyError:
-            pass
+    def get(self, key: str, refresh: bool = False) -> RedisCluster[str] | StrictRedis[str]:
```

Comment:
Instead of adding boolean flag parameters, you could have a `.refresh(key)` method that refreshed the cluster, updated the cluster map and returned the refreshed instance. A rough get/refresh method could look like

```python
def get(self, key):
    try:
       return self._clusters[key]
    except KeyError:
        pass
    return self.refresh(key)

def refresh(self, key):
    cfg = self._options_manager.get("redis.clusters", {}).get(key)
    if cfg is None:
        raise KeyError(f"Invalid cluster name: {key}")
    if cfg.get("is_redis_cluster", False):
        raise KeyError("Invalid cluster type, expected rb cluster")
    ret = self._clusters[key] = self._factory(**cfg)
    return ret
```